.scalafix.conf

```diff
@@ -34,3 +104,127 @@ Disable.ifSynthetic = [
+//      "scalaz/zio/ZIOFunctions.fail"
+//      "scalaz/zio/ZIOFunctions.halt"
+//    ]
+//    symbols = [
+//      {
+//        # This takes the approach of banning everything and then blessing things
+//        # we trust. It would be entirely feasible to do it the other way and
+//        # list the explicit list of things that we need to go through IO, but it
+//        # is less safe. We can have another rule if something in an "excludes"
+//        # glob catches unsafe things.
+//        #
+//        # If something is referentially transparent but not total, instead of
+//        # adding to this excludes list, add it to the includes list of the
+//        # scalaz.Maybe.attempt safeBlock. If it can return `null`, add it to a
+//        # scalaz.Maybe.fromNullable safeBlock.
+//        regex = {
+//          includes = [
+//            "^\\Qjava/\\E.*$"
+//            #"^.*$"
+//          ]
+//          excludes = [
+//            "^fommil.*$"
+//            "^scalaz.*$"
+//            "^\\Qjava/lang/String#\\E.*$"
+//            "^\\Qjava/time/Instant#\\E.*$"
+//            "^\\Qjava/time/ZonedDateTime#\\E.*$"
+//            "^\\Qjava/time/temporal/ChronoUnit#\\E.*$"
+//            "^\\Qjava/net/URI#\\E.*$"
+//            "^\\Qjava/net/URLEncoder#\\E.*$"
+//            "^\\Qjava/net/URLDecoder#\\E.*$"
+//            # primitives are not supported: https://github.com/scalameta/scalameta/issues/1488
+//
+//            # WORKAROUND https://github.com/scalacenter/scalafix/issues/715
+//            "^\\Qjava/\\E$"
+//            "^\\Qjava/lang/\\E$"
+//            "^\\Qjava/time/\\E$"
+//            "^\\Qjava/net\\E$"
+//          ]
+//        }
+//        message = "Untrusted third party library must be called from IO, or blessed in scalafix.conf"
+//      }
+//    ]
+//  }
+//  {
+//    safeBlocks = [
+//      "scalaz/`\/`.fromTryCatchNonFatal",
+//      "scalaz/`\/`.fromTryCatchThrowable",
+//      "scalaz/Maybe.attempt",
+//      "scalaz/Maybe.fromTryCatchNonFatal",
+//      "scalaz/Maybe.fromTryCatchThrowable"
+//    ]
+//    symbols = [
+//      {
+//        regex = {
+//          includes = [
+//            # should live in a common file so we don't need to duplicate
+//            "^\\Qjava/net/URLEncoder#\\E.*$"
+//            "^\\Qjava/net/URLDecoder#\\E.*$"
+//          ]
+//          excludes = [
+//          ]
+//        }
+//        message = "Deterministic method is not total, must be called via Maybe.attempt, \/.fromTryCatchNonFatal, etc."
+//      }
+//    ]
+//  }
+//]
+
+DisableSyntax {
+  noAsInstanceOf = true
+  noContravariantTypes = true
+  noCovariantTypes = true
+//  noDefaultArgs = true
+  noFinalVal = true
+  noFinalize = true
+  noImplicitConversion = true
+  noImplicitObject = true
+  noIsInstanceOf = true
+  noNulls = true
+  noReturns = true
+  noSemicolons = true
+  noTabs = true
+  noThrows = true
+  noUniversalEquality = true
+//  noValInAbstract = true
+  noValPatterns = true
```

Comment:
Probably not in tests

Reply:
unfortunately I don't know to enforce a rule only in `Compile`, and ignore it in `Test` :slightly_frowning_face: